debian/libmiral4.symbols

```diff
@@ -259,7 +259,6 @@ libmiral.so.4 libmiral4 #MINVER#
  (c++)"miral::WindowManagerTools::window_at(mir::geometry::Point) const@MIRAL_3.0" 3.0.0
  (c++)"miral::WindowManagerTools::~WindowManagerTools()@MIRAL_3.0" 3.0.0
  (c++)"miral::WindowSpecification::WindowSpecification()@MIRAL_3.0" 3.0.0
- (c++)"miral::WindowSpecification::WindowSpecification(mir::scene::SurfaceCreationParameters const&)@MIRAL_3.0" 3.0.0
```

Comment:
This (and `miral::WindowSpecification::update(mir::scene::SurfaceCreationParameters&)`) is a ABI break.

We can probably get away with this as there is no sane way a downstream could be using these functions. (This would be clearer if they had been made private, or non-member `friend` functions originally.)

Reply:
- Restored the dropped functions and replaced their implementation with a deprecation warning.
- Created internal `WindowSpecification` header and source files
- Replaced the new `apply_to` function with a standalone internal `make_surface_spec()` function.